### PR TITLE
fix: Panic in IsInvalidParameterValueError

### PR DESCRIPTION
**Pull request by irmatov**
Closes https://github.com/cloudquery/cloudquery-issues/issues/381

### CODE REVIEW

**Comment by yevgenypats** [comment]
Does it fix it? why behaviour is different with explicit variable here?

File: client/helpers.go [hunk 1]
```diff
@@ -302,7 +302,7 @@ func isNotFoundError(err error) bool {
 
 func IsInvalidParameterValueError(err error) bool {
 	var apiErr smithy.APIError
-	if errors.As(err, &apiErr); apiErr.ErrorCode() == "InvalidParameterValue" {
+	if ok := errors.As(err, &apiErr); ok && apiErr.ErrorCode() == "InvalidParameterValue" {
```
Comment (disq) [inline client/helpers.go:305]:
```suggestion
	if errors.As(err, &apiErr) && apiErr.ErrorCode() == "InvalidParameterValue" {
```
simpler?

**Comment by yevgenypats** [comment]
Looks like the original way is preferable per their docs - https://pkg.go.dev/errors

can you elaborate or add a test to make sure it fix it?

File: client/helpers.go (hunk 1 above)
Comment (yevgenypats) [reply client/helpers.go:305]: yes this should be the way to go and the suggested one

Comment (irmatov) [reply client/helpers.go:305]: @yevgenypats You're missing that current code does not test a result of `errors.As`, so if it is `false` then `apiErr` is nil thus panic.

**Review by disq — APPROVED** [review]

**Comment by erezrokah** [comment]
Thanks for the fix @irmatov and the review @disq and @yevgenypats

**Pull request merged**